### PR TITLE
[Feature branch] openpifpaf integration

### PR DESCRIPTION
**Pull request by corey-nm**
Base feature  branch

Manual tests:
1. Ran the training commands listed in the README with yaml file shown in readme and verified sparsity levels changed via log output
2. Ran the export commands listed in the README with the checkpoint from training command & verified output is runnable with deepsparse.benchmark
3. Ran the training command with --checkpoint instead of --basenet arg from a non-finished training epoch to verify resume logic
4. Ran the training command with --checkpoint instead of --basenet from a finished training epoch to verify checkpoint logic

### CODE REVIEW

File: src/sparseml/pytorch/sparsification/pruning/modifier_pruning_obs.py [hunk 1]
```diff
@@ -518,7 +518,7 @@ def mask_update(self, masks: List[Tensor], mask_diffs: List[Tensor]):
                     obs_updates[i] = (
                         self._finvs[i]
                         .mul(
-                            self._block_finv_w[i].view(-1)
+                            self._block_finv_w[i].reshape(-1)
```
Comment (dbogunowicz) [inline src/sparseml/pytorch/sparsification/pruning/modifier_pruning_obs.py:521]: Isn't "view" more efficient?

File: setup.py [hunk 2]
```diff
@@ -62,12 +62,12 @@
 
 _onnxruntime_deps = ["onnxruntime>=1.0.0"]
 _pytorch_deps = [
-    "torch>=1.1.0,<=1.12.1",
+    "torch>=1.1.0,<=1.13.0",
```
Comment (dbogunowicz) [inline setup.py:65]: Sorry, I am not up-to-date, 1.13 update for sparseml is incoming right?

**Review by dbogunowicz — DISMISSED** [review]

File: src/sparseml/pytorch/sparsification/pruning/modifier_pruning_obs.py (hunk 1 above)
Comment (dbogunowicz) [reply src/sparseml/pytorch/sparsification/pruning/modifier_pruning_obs.py:521]: Oh wait, I think you need to clean up your commit history right

File: setup.py [hunk 3]
```diff
@@ -76,13 +76,15 @@
...
+_open_pif_paf_deps = ["openpifpaf==0.13.6"]
+
 _dev_deps = [
     "beautifulsoup4==4.9.3",
     "black==21.5b2",
     "flake8==3.9.2",
     "isort==5.8.0",
     "m2r2~=0.2.7",
-    "mistune==0.8.4",
+    "mistune<3, >=2.0.3'",
```
Comment (corey-nm) [inline setup.py:87]:
```suggestion
    "mistune<3,>=2.0.3",
```

File: setup.py [hunk 4]
```diff
@@ -211,6 +213,15 @@ def _setup_entry_points() -> Dict:
         "sparseml.recipe_template=sparseml.pytorch.recipe_template.cli:main"
     )
 
+    # pose detection entrypoint
+
+    entry_points["console_scripts"].extend(
+        [
+            "sparseml.openpifpaf.train=" "sparseml.openpifpaf.train:main",
+            "sparseml.openpifpaf.export_onnx=" "sparseml.openpifpaf.export:main",
```
Comment (corey-nm) [inline setup.py:221]:
```suggestion
            "sparseml.openpifpaf.train=sparseml.openpifpaf.train:main",
            "sparseml.openpifpaf.export_onnx=sparseml.openpifpaf.export:main",
```

File: src/sparseml/openpifpaf/export.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+):
+    if one_shot:
+        manager = ScheduledModifierManager.from_yaml(one_shot)
+        manager.apply(model)
+
+    image_size_warning(model.base_net.stride, input_width, input_height)
+
+    # configure
+    openpifpaf.network.heads.CompositeField3.inplace_ops = False
+    openpifpaf.network.heads.CompositeField4.inplace_ops = False
```
Comment (KSGulin) [inline src/sparseml/openpifpaf/export.py:109]: nit and not needed at this stage, but I think additional comments around some of the openpifpaf calls/modifications would be useful, as they seem pretty opaque

**Review by KSGulin — APPROVED** [review]
Great stuff. Love that it fits within sparseml without need for a fork

**Review by dbogunowicz — APPROVED** [review]
🚀

**Review by bfineran — APPROVED** [review]
🎆

**Pull request merged**